### ISSUE TITLE
Re-enable launching external app-links from web content.

### ISSUE THREAD

**Issue by jhugman** [mozilla-mobile/android-components#3687]
This depends on [geckoview#1555337][1].

This should be a matter of flipping the default value for the constructor flag for `AppLinksFeature`.

Requires #3691 
Requires #3690 

[1]: https://bugzilla.mozilla.org/show_bug.cgi?id=1555337

**Comment by jhugman** [comment]
I'd like to progress this, despite geckoview#1555337.

webcompat is becoming the dominant issue here.

**Comment by cpeterso** [comment]
The GV bug is in the GV team's October sprint backlog: https://bugzilla.mozilla.org/show_bug.cgi?id=1555337#c20

**Comment by cpeterso** [comment]
The GV bug fix landed yesterday, so I'm removing this issue's `needs:gv` and `blocked` labels. GV has a new `LoadRequest#hasUserGesture` API.

https://bugzilla.mozilla.org/show_bug.cgi?id=1555337#c26

**Comment by liuche** [comment]
@rocketsroger when you said you're working on AC issues related to app links https://github.com/mozilla-mobile/fenix/issues/3177 is this the one, or are there others? I'd like to link any AC dependencies that need to be finished before Fenix can start on that work so eng knows when it's available to be worked on.

**Comment by rocketsroger** [comment]
@liuche I'm working on others.  This one should be closed.  This is a duplicate of https://github.com/mozilla-mobile/android-components/issues/3857.  There will be a need to create an Fenix issue soon to implement a setting preference for launching app link in third party apps.  However, this will have to be done by me because the a-c change will be API breaking.

**Comment by rocketsroger** [comment]
DUP https://github.com/mozilla-mobile/android-components/issues/3857
